### PR TITLE
feat: Enable to build native libs on Linux

### PR DESCRIPTION
**Pull request by Jklawreszuk**
# PR Details

## Description

The goal is to be able to build projects that are using native c++ code (e.g. Stride.Audio, Stride.Rendering etc.)

## Related Issue

https://github.com/stride3d/stride/issues/360

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] **I have built and run the editor to try this change out.**

### CODE REVIEW

**Comment by Jklawreszuk** [comment]
I guess my work is ready for review :smiley: 
Please take a note here you can only build Linux native libraries, but this is good enough to run and test most Stride project and clang further improvements can be made in other PRs

File: sources/native/Stride.Native.targets [hunk 1]
```diff
@@ -39,9 +45,9 @@
   
   <!-- Define default CPU architectures -->
   <ItemGroup Condition="'$(TargetFramework)' == '$(StrideFramework)'">
-    <StrideNativeCPU Include="win-x64;win-x86" LibraryExtension=".dll" LibraryRuntime="win" />
+    <StrideNativeCPU Condition="$([MSBuild]::IsOSPlatform('Windows'))" Include="win-x64;win-x86" LibraryExtension=".dll" LibraryRuntime="win" />
     <StrideNativeCPU Include="linux-x64"  LibraryExtension=".so" LibraryRuntime="linux" />
-    <StrideNativeCPU Include="osx-x64"  LibraryExtension=".dylib" LibraryRuntime="osx" />
+    <StrideNativeCPU Condition="$([MSBuild]::IsOSPlatform('Windows'))" Include="osx-x64"  LibraryExtension=".dylib" LibraryRuntime="osx" />
```
Comment (Eideren) [inline sources/native/Stride.Native.targets:50]: Did you mean to use windows here ? and at line 22 ?

Comment (Jklawreszuk) [reply sources/native/Stride.Native.targets:50]: @Eideren I had to restrict generating dlls and dylib only on Windows as a **temporary** workaround.
Two reassons: `CompileNativeClang_Windows` target tries to build using vcxproj - Visual C++ is supported only by msbuild NET FM and `CompileNativeClang_macOS` uses darwin-ld.exe (???????) - I didn't figure out where the heck I could find apple ld. Only clue I found is  [LLVM apple's fork ](https://github.com/apple/llvm-project).
To sum up, I don't want to break things yet but only allow to extend building for Linux for now.

Maybe I should add a comment here?

Comment (Eideren) [reply sources/native/Stride.Native.targets:50]: Yes, please do

Comment (Jklawreszuk) [reply sources/native/Stride.Native.targets:51]: Done! Hope that be useful for others :smile:

**Comment by Eideren** [comment]
Thanks !
